oauth2/fosite_store_helpers.go

```diff
@@ -491,14 +492,17 @@ func testHelperFlushTokensWithLimitAndBatchSize(x InternalRegistry, limit int, b
 		}
 
 		require.NoError(t, m.FlushInactiveAccessTokens(ctx, time.Now(), limit, batchSize))
+		var notFoundCount, foundCount int
 		for i := range requests {
-			_, err := m.GetAccessTokenSession(ctx, requests[i].ID, ds)
-			if i >= limit {
-				require.NoError(t, err)
+			if _, err := m.GetAccessTokenSession(ctx, requests[i].ID, ds); err == nil {
+				foundCount++
 			} else {
-				require.Error(t, err)
+				require.ErrorIs(t, err, fosite.ErrNotFound)
+				notFoundCount++
 			}
 		}
+		assert.Equal(t, limit, notFoundCount, "should have deleted %d tokens", limit)
+		assert.Equal(t, totalCount-limit, foundCount, "should have found %d tokens", totalCount-limit)
```

Comment:
Because we delete the access keys up to `limit` based on the signature, and we now always hash the signature, this will give us a random ordering of deleted vs non-deleted tokens.